src/app/public/plugin-resources/code-examples/flyout/custom-headers/flyout-demo.component.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  Component
+} from '@angular/core';
+
+import {
+  SkyFlyoutConfig,
+  SkyFlyoutInstance,
+  SkyFlyoutService
+} from '@skyux/flyout';
+
+import {
+  FlyoutDemoFlyoutComponent
+} from './flyout-demo-flyout.component';
+
+@Component({
+  selector: 'app-flyout-demo',
+  templateUrl: './flyout-demo.component.html'
+})
+export class FlyoutDemoComponent {
+
+  public flyout: SkyFlyoutInstance<any>;
+
+  constructor(
+    private flyoutService: SkyFlyoutService
+  ) { }
+
+  public openFlyoutWithIterators(): void {
+    const flyoutConfig: SkyFlyoutConfig = {
+      showIterator: true
+    };
+    this.flyout = this.flyoutService.open(FlyoutDemoFlyoutComponent, flyoutConfig);
+
+    this.flyout.iteratorNextButtonClick.subscribe(() => {
+      alert('Next iterator button clicked!');
+    });
+
+    this.flyout.iteratorPreviousButtonClick.subscribe(() => {
+      alert('Prvious iterator button clicked!');
+    });
+
+    this.flyout.closed.subscribe(() => {
+      this.flyout = undefined;
+    });
+  }
+
+  public openFlyoutWithRoutePermalink(): void {
+    const flyoutConfig: SkyFlyoutConfig = {
+      permalink: {
+        label: 'Go to Components page',
+        route: {
+          commands: ['/components'],
+          extras: {
+            fragment: 'helloWorld',
+            queryParams: {
+              foo: 'bar'
+            }
+          }
+        }
+      }
+    };
+    this.flyout = this.flyoutService.open(FlyoutDemoFlyoutComponent, flyoutConfig);
+
+    this.flyout.closed.subscribe(() => {
+      this.flyout = undefined;
+    });
+  }
+
+  public openFlyoutWithUrlPermalink(): void {
+    const flyoutConfig: SkyFlyoutConfig = {
+      permalink: {
+        label: `Visit blackbaud.com`,
```

Comment:
In the demo, when I click this button, it opens blackbaud.com. But in Stackblitz, it just closes the flyout. Is this the expected behavior?

Reply:
Yeah, stackblitz prevents the link from firing for some reason. It also doesn't work for router params.